src/main.rs

```diff
@@ -312,240 +326,264 @@ async fn handle_nwc_params(
     mut lnd: LndLightningClient,
 ) -> anyhow::Result<()> {
     let mut d_tag: Option<Tag> = None;
-    let content = match params {
-        RequestParams::PayInvoice(params) => {
-            d_tag = params.id.map(Tag::Identifier);
-
-            let invoice = Bolt11Invoice::from_str(&params.invoice)
-                .map_err(|_| anyhow!("Failed to parse invoice"))?;
-            let msats = invoice
-                .amount_milli_satoshis()
-                .or(params.amount)
-                .unwrap_or(0);
-
-            let error_msg = if config.max_amount > 0 && msats > config.max_amount * 1_000 {
-                Some("Invoice amount too high.")
-            } else if config.daily_limit > 0
-                && tracker.lock().await.sum_payments() + msats > config.daily_limit * 1_000
-            {
-                Some("Daily limit exceeded.")
-            } else {
-                None
-            };
-
-            // verify amount, convert to msats
-            match error_msg {
-                None => {
-                    match pay_invoice(invoice, lnd, method).await {
-                        Ok(content) => {
-                            // add payment to tracker
-                            tracker.lock().await.add_payment(msats);
-                            content
-                        }
-                        Err(e) => {
-                            error!("Error paying invoice: {e}");
-
-                            Response {
-                                result_type: method,
-                                error: Some(NIP47Error {
-                                    code: ErrorCode::InsufficientBalance,
-                                    message: format!("Failed to pay invoice: {e}"),
-                                }),
-                                result: None,
+
+    let content;
+    if config.recv_only() && !RECV_ONLY_METHODS.contains(&method) {
+        content = Response {
+            result_type: method,
+            error: Some(NIP47Error {
+                code: ErrorCode::Restricted,
+                message: "Method not allowed.".to_string(),
+            }),
+            result: None,
+        };
+    } else {
```

Comment:
I wanted to use early return here but the code below (starting at L588) needs to execute which would require some refactoring afaict